include/pybind11/cast.h

```diff
@@ -876,6 +876,38 @@ struct handle_type_name {
     static constexpr auto name = const_name<T>();
 };
 template <>
+struct handle_type_name<object> {
+    static constexpr auto name = const_name("object");
+};
+template <>
+struct handle_type_name<list> {
+    static constexpr auto name = const_name("list");
+};
+template <>
+struct handle_type_name<dict> {
+    static constexpr auto name = const_name("dict");
+};
+template <>
+struct handle_type_name<anyset> {
+    static constexpr auto name = const_name("set");
```

Comment:
According to https://docs.python.org/3/c-api/set.html `anyset` is an alias for "set or fronzenset".

The contexts where rendering it as `set | frozenset` might lead to trouble are pretty exotic, I think it's ok to render as union type: 

```suggestion
    static constexpr auto name = const_name("set | frozenset");
```

Reply:
Done: ace70b077fed43e0bf9bcc90bf84587cade3ad5a

I see `Union[...]` is used in multiple places.

Could it be better to use `Union[set, frozenset]` for consistency?

Reply:
Yes, let's use `Union` for consistency

Reply:
Done: 7c8991a0a0d14ec325714d994689e723a8affdba